content/blog/2023/01/camel-k-yaks/index.md

```diff
@@ -0,0 +1,311 @@
+```
+
+Let’s walk through the test step by step.
+First of all the feature file uses the usual Given-When-Then BDD syntax to give context, describe the actions and verify the outcome.
+Each step calls a specific YAKS action that is provided out of the box by the framework.
+The user is able to choose from a [huge set of steps](https://github.com/citrusframework/yaks/tree/main/java/steps) that
+automatically perform actions like sending/receiving Http requests/responses, starting [Testcontainers](https://www.testcontainers.org/),
+running Camel routes, connecting to a database, publishing events on Kafka or Knative brokers and many more.
+
+In the first scenario the test automatically prepares some required infrastructure.
+The YAKS test starts a Localstack [Testcontainer](https://www.testcontainers.org/) to have an AWS S3 test instance running (`Given start LocalStack container`).
+Then the test loads and starts the Camel K integration under test (`When load Camel K integration fruit-service.groovy`) and waits for it to properly start.
+In local testing this step starts the Camel K integration using JBang.
+Later the post will also run the test in a Kubernetes environment.
+
+Now the infrastructure is up and running and the test is able to load the fruit model object as Http request body
+(`Given HTTP request body: yaks:readFile('pineapple.json')`) and invoke the Camel K service (`When send POST /fruits`).
+The test waits for the Http response and verifies its 200 OK status.
+
+In the last step the test verifies that the fruit object has been added to the right AWS S3 bucket (medium-sugar).
+As YAKS itself is not able to connect to AWS S3 the test uses Apache Camel for this step.
+The test creates a Camel context, loads a AWS client and connects to AWS S3 with a temporary Camel route
+(`Given receive Camel exchange from("aws2-s3://medium-sugar?amazonS3Client=#amazonS3Client&deleteAfterRead=true")`).
+With this Apache Camel integration YAKS is able to use the complete 300+ Camel components for sending and receiving messages
+to various messaging transports. The Camel exchange body should be the same fruit model object (`yaks:readFile('pineapple.json'`)
+as posted in the initial Http request.
+
+YAKS uses the powerful message payload validation capabilities provided by Citrus for this message content verification.
+The validation is able to compare message contents of type XML, Json, plaintext and many more.
+
+This completes the test case. You can now run this test with Cucumber and JUnit for instance.
+The easiest way though to directly run tests with YAKS is to use the [YAKS command line client](https://github.com/citrusframework/yaks/releases).
+You do not need to set up a whole project with Maven dependencies and so on.
+Just write the test file and run with:
+
+```shell script
+$ yaks run fruit-service.feature --local
+```
+
+You should see some log output like this:
+
+```
+INFO | 
```

Comment:
@christophd try removing trailing whitespace like this:

```suggestion
INFO |
```

Reply:
ah! my IDE does not touch these pre-formatted code blocks. thx @zregvart